tools/moduleversions/internal/helper/helpers.go

```diff
@@ -0,0 +1,163 @@
+	URL       string
+	FetchTime time.Time
+}
+
+// VerifyVersionOnDocumentationSite checks if the specified version exists for the given channel
+// on the deckhouse.ru documentation site
+func VerifyVersionOnDocumentationSite(docURL, expectedChannel, expectedVersion string) error {
+	info := &DocumentationSiteInfo{
+		Channels:  map[string]string{},
+		URL:       docURL,
+		FetchTime: time.Now(),
+	}
+
+	client := &http.Client{
+		Timeout: httpTimeout,
+	}
+
+	resp, err := client.Get(docURL)
+	if err != nil {
+		return fmt.Errorf("failed to fetch documentation URL %s: %w", docURL, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected status code %d for URL %s", resp.StatusCode, docURL)
+	}
+
+	doc, err := goquery.NewDocumentFromReader(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to parse HTML from %s: %w", docURL, err)
+	}
+
+	// Parse channel-version pairs from the documentation site
+	doc.Find(".submenu-item").Each(func(i int, s *goquery.Selection) {
+		channel := s.Find(".submenu-item-channel").Text()
+		channel = strings.TrimSpace(channel)
+		channel = GetChannel(channel)
+
+		version := s.Find(".submenu-item-release").Text()
+		version = GetSemVer(version)
+
+		if channel != "" && version != "" {
+			info.Channels[channel] = version
+		}
+	})
+
+	// Verify the expected version for the expected channel
+	foundVersion, exists := info.Channels[expectedChannel]
+	if !exists {
+		return fmt.Errorf("channel %s not found on documentation site %s", expectedChannel, docURL)
+	}
+
+	if foundVersion != expectedVersion {
+		return fmt.Errorf("version mismatch on documentation site: expected %s for channel %s, got %s", expectedVersion, expectedChannel, foundVersion)
+	}
+
+	fmt.Printf("Found version %s for channel %s on documentation site\n", foundVersion, expectedChannel)
+	return nil
+}
+
+// verifyVersionInEdition checks if the specified version exists for the given channel
+// on the releases.deckhouse.io page for a specific edition
+func VerifyVersionInEdition(editionURL, channel, expectedVersion, moduleName string) (bool, error) {
+	client := &http.Client{
+		Timeout: httpTimeout,
+	}
+
+	resp, err := client.Get(editionURL)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch edition URL %s: %w", editionURL, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return false, fmt.Errorf("unexpected status code %d for URL %s", resp.StatusCode, editionURL)
+	}
+
+	doc, err := goquery.NewDocumentFromReader(resp.Body)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse HTML from %s: %w", editionURL, err)
+	}
+
+	var (
+		index      int
+		webVersion string
+	)
+
+	switch channel {
+	case "alpha":
+		index = channelAlphaIndex
+	case "beta":
+		index = channelBetaIndex
+	case "early-access":
+		index = channelEarlyAccessIndex
+	case "stable":
+		index = channelStableIndex
+	case "rock-solid":
+		index = channelRockSolidIndex
```

Comment:
I think it would be better to have this boilerplate replaced by a map[string]int, which would resolve an index by the name of the channel.
And here you would have something like this:

```suggestion
index = ChannelMap[channel]
```